spot-contracts/contracts/_interfaces/IPerpetualTranche.sol

```diff
@@ -38,11 +38,29 @@ interface IPerpetualTranche is IERC20Upgradeable {
     /// @notice Deposits tranche tokens into the system and mint perp tokens.
     /// @param trancheIn The address of the tranche token to be deposited.
     /// @param trancheInAmt The amount of tranche tokens deposited.
-    function deposit(ITranche trancheIn, uint256 trancheInAmt) external;
+    function deposit(ITranche trancheIn, uint256 trancheInAmt) external returns (uint256);
 
     /// @notice Burn perp tokens and redeem the share of reserve assets.
     /// @param perpAmtBurnt The amount of perp tokens burnt from the caller.
-    function redeem(uint256 perpAmtBurnt) external;
+    function redeem(uint256 perpAmtBurnt)
+        external
+        returns (IERC20Upgradeable[] memory tokensOut, uint256[] memory tokenOutAmts);
+
+    struct RolloverPreview {
```

Comment:
Minor, but wonder if this should be renamed since it's also a rollover return data payload.
If we do, maybe save it for just a rename PR.

Reply:
We can rename it to `RolloverData`